olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -341,6 +341,33 @@ class DATASERVICE_READ_API VersionedLayerClient final {
    */
   bool RemoveFromCache(const geo::TileKey& tile);
 
+  /**
+   * @brief Fetches data of a tile or its closest ancestor.
+   *
+   * @param request The `TileRequest` instance that contains a complete set
+   * of request parameters.
+   * @note CacheWithUpdate fetch option is not suported.
+   * @param callback The `DataResponseCallback` object that is invoked if
+   * the `DataResult` object is available or an error is encountered.
+   *
+   * @return A token that can be used to cancel this request.
+   */
+  client::CancellationToken GetAggregatedData(TileRequest request,
+                                              DataResponseCallback callback);
```

Comment:
Probably the DataResponseCallback is not enough. We may need a separate response type in which we also say which TileKey we found and the data belongs too.

Reply:
How about data handle? I think its nice to know it too.

Reply:
This might confuse the user in thinking the data handle is constant and he can use it always.